oletools/olevba.py

```diff
@@ -1937,58 +1938,57 @@ def __init__(self, ole, vba_root, project_path, dir_path, relaxed=True):
                 else:
                     referencecontrol_reserved3 = check2
 
-                self.check_value('REFERENCECONTROL_Reserved3', 0x0030, referencecontrol_reserved3)
+                self.check_value(REFERENCECONTROL + '_Reserved3', 0x0030, referencecontrol_reserved3)
                 referencecontrol_sizeextended = struct.unpack("<L", dir_stream.read(4))[0]
                 referencecontrol_sizeof_libidextended = struct.unpack("<L", dir_stream.read(4))[0]
                 referencecontrol_libidextended = dir_stream.read(referencecontrol_sizeof_libidextended)
+                log.debug(REFERENCECONTROL + ' lib id extended: %s' % unicode2str(self.decode_bytes(referencecontrol_libidextended)))
                 referencecontrol_reserved4 = struct.unpack("<L", dir_stream.read(4))[0]
                 referencecontrol_reserved5 = struct.unpack("<H", dir_stream.read(2))[0]
                 referencecontrol_originaltypelib = dir_stream.read(16)
                 referencecontrol_cookie = struct.unpack("<L", dir_stream.read(4))[0]
                 unused = referencecontrol_sizeextended
-                unused = referencecontrol_libidextended
                 unused = referencecontrol_reserved4
                 unused = referencecontrol_reserved5
                 unused = referencecontrol_originaltypelib
                 unused = referencecontrol_cookie
                 continue
 
             if check == 0x000D:
-                # REFERENCEREGISTERED
+                REFERENCEREGISTERED = "REFERENCEREGISTERED"
                 # Specifies a reference to an Automation type library.
                 referenceregistered_id = check
                 referenceregistered_size = struct.unpack("<L", dir_stream.read(4))[0]
                 referenceregistered_sizeof_libid = struct.unpack("<L", dir_stream.read(4))[0]
-                referenceregistered_libid = dir_stream.read(referenceregistered_sizeof_libid)
-                log.debug('REFERENCE registered lib id: %s' % unicode2str(self.decode_bytes(referenceregistered_libid)))
+                referenceregistered_libid = unicode2str(self.decode_bytes(dir_stream.read(referenceregistered_sizeof_libid)))
```

Comment:
Only wanted to call the decode once. Result is used in the log and also sent to the references list